### ISSUE TITLE
Find a way to check ``ARITH`` usage

### ISSUE THREAD

**Issue by MCJOHN974** [near/wasmtime#71]
zkasm prover don't fail if you forget to write an ``ARITH`` after free input. Currently we check if we care about all free inputs just using code reviews in the PRs. I think we want to have some tool which will automatically catch if we miss to check a free input

**Comment by MCJOHN974** [comment]
As was discussed at one of meeting only way to check ``ARITH`` usage is manual human review. I don't like such approach, and what about building some static-analysis tool which will help validate ``ARITH`` usage in Stage-2? I don't think we are able to build tool which will work in 100% cases in reasonable time, but I believe it is quite easy to build a tool which will work in most part of cases. I think most difficulties will be in cases like:

```
<external export>
CJMP label1 label2
...
label1:
ARITH1
label2:
ARITH2
```

**Comment by MCJOHN974** [comment]
@akashin @nagisa WDYT about adding it as a task for stage-2, does it worth it or we have enough more important tasks?

**Comment by akashin** [comment]
> @akashin @nagisa WDYT about adding it as a task for stage-2, does it worth it or we have enough more important tasks?

I agree that automated validation/detailed auditing is out of scope of stage 1 and we can move it to stage 2 or 3.
We still should do our best to make sure our code is designed in a sound matter from the ZK perspective (e.g. not taking shortcuts of computing the value as free input and never validating it).

I also suggest asking Polygon about tooling that they already have for this - this doesn't seem like a problem that is exclusive to our use-case and I suspect they already did some work on this front.
